src/main.rs

```diff
@@ -348,6 +372,40 @@ async fn forward_request(
     } else {
         None
     };
+
+    let decoded_body = decode_body(&http_response.body, encoding.clone());
+    let body_valid = match (certificate.clone(), tree.clone()) {
```

Comment:
Is some kind of zip-bomb attack possible here, so that a max size needs to be prescribed?

Reply:
Thank you! Added `MAX_BYTES_SIZE_TO_DECOMPRESS` param

Reply:
can we go further actually and not pass the body around, but just the sha?

What I'm thinking is we just decode 1K chunk of the body, add that 1K to the sha. Lather rinse and repeat up to EOF or `MAX_BYTES_SIZE_TO_DECOMPRESS` (or maybe do `0..MAX_CHUNKS_TO_DECOMPRESS`). Then return the `body_sha` and pass that into `validate_body`

Reply:
Great idea!
Left the possibility of partial hash calculation